model/src/main/proto/exploration.proto

```diff
@@ -0,0 +1,145 @@
+syntax = "proto3";
+
+import "subtitled_html.proto";
+
+package model;
+
+option java_package = "org.oppia.app.model";
+option java_multiple_files = true;
+
+// Data for a single exploration.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeExploration.kt
+message Exploration {
+  repeated State states = 1;
+  repeated ParamChange param_changes = 2;
+  repeated ParamSpec param_specs = 3;
+  string initStateName = 4;
+  string objective = 5;
+  bool correctness_feedback_enabled = 6;
+  string title = 7;
+  string language_code = 8;
+}
+
+// Data for a param change.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeParamChange.kt
+message ParamChange {
+  string generator_id = 1;
+  string name = 2;
+  CustomizationArgs customization_args = 3;
+}
+
+// Data for a param spec.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeParamSpec.kt
+message ParamSpec {
+  string obj_type = 1;
+}
+
+// Data for a single state
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeState.kt
+message State {
+  // Mapping from content_id to a VoiceoverMapping
+  map<string, VoiceoverMapping> recorded_voiceovers = 1;
+  SubtitledHtml content = 2;
+  // Mapping from content_id to a TranslationMapping
+  map<string, TranslationMapping>written_translations = 3;
+  repeated ParamChange param_changes = 4;
+  string classifier_model_id = 5;
+  Interaction interaction = 6;
+  bool solicit_answer_details = 7;
+}
+
+// Data for customization args.
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeCustomizationArgs.kt
+message CustomizationArgs {
+  bool parse_with_jinja = 1;
+  string value = 2;
+}
+
+// Mapping from language_code to a VoiceOver
+message VoiceoverMapping {
+  map<string, Voiceover> voiceover_mapping = 1;
+}
+
+// Data for a single voiceover
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeVoiceover.kt
+message Voiceover {
+  int64 file_size_bytes = 1;
+  bool needs_update = 2;
+  string file_name = 3;
+}
+
+// Data for a single interaction
+// Maps from: data/src/main/java/org/oppia/data/backends/gae/model/GaeInteractionInstance.kt
+message Interaction {
+  string id = 1;
+  repeated AnswerGroup answer_groups = 2;
+  Solution solution = 3;
+  repeated AnswerGroup confirmed_unclassified_answers = 4;
+  Hint hint = 5;
+  Outcome default_outcome = 6;
+  map<string, CustomizationArgs> customization_args = 7;
```

Comment:
Here & elsewhere: please add a comment documenting what the key corresponds to since it's just a string. Without this, it's hard to know how to use the map correctly.

Reply:
Done